docs/docs/setup/web-wasm-setup.md

```diff
@@ -28,3 +28,73 @@ For more details on building and integrating these circuits, refer to [`mopro-wa
    Open a web browser and visit the test page at the default url: `http://localhost:3000`.
    
    Check the results displayed in the browser console or user interface.
+
+
+## Integrating a New Circuit Implementation
+
+This section assumes the existence of a user-defined circuit implementation based on the [PSE Halo2](https://github.com/privacy-scaling-explorations/halo2). 
+
+Note that there are multiple Halo2 implementations (e.g., Zcash, PSE, Axiom). Mopro primarily supports the [PSE Halo2](https://github.com/privacy-scaling-explorations/halo2), which is a Plonk backend and works well with [wasm-bindgen-rayon](https://github.com/RReverser/wasm-bindgen-rayon). Refer to the [README](https://github.com/zkmopro/mopro/tree/main/mopro-wasm#introduction-to-wasm-compilation-with-halo2) of mopro-wasm for information about compatibility between Halo2 and Wasm.
+
+Follow these steps to update and build the user-defined circuit implementation for Wasm:
+
+1. **Update `Cargo.toml` in 'mopro-wasm'**
+
+    The customized circuit crate should be added as a dependency manually, as illustrated below:
+
+      ```toml
+      [dependencies]
+      ...
+      gemini-fibonacci = { package = "gemini-fibonacci", git = "https://github.com/sifnoc/plonkish-fibonacci-sample.git", optional = true }
+      my-halo2-circuit = { ... }
+      ```
+
+2. **Create Wrapper Functions for Generate/Verify proof method**:
+
+   To compile Wasm code with the circuit, wrapper functions for generating and verifying proof methods in the user-defined circuit implementation must be created in `mopro-wasm/src/lib.rs`, using the example structure provided below:
+
+   ```rust
+   use my_halo2_circuit;
+
+   #[wasm_bindgen]
+   pub fn generate_proof(input: JsValue) -> Result<JsValue, JsValue> {
+      // function implementations with `my-halo2-circuit`
+      let proof = my_halo2_circuit::generate_proof(parsed_input);
+      to_value(...)
+   }
+
+   #[wasm_bindgen]
+   pub fn verify_proof(proof: JsValue, public_inputs: JsValue) -> Result<JsValue, JsValue> {
+      // function implementations with `my-halo2-circuit`
+      let result = my_halo2_circuit::verify_proof(parsed_proof, parsed_public_input);
+      to_value(...)
+   }
+   ```
+
+3. **Build the Wasm Package**
+
+    The following command builds the wasm package in "mopro-wasm":
+
+   ```shell
+   mopro-wasm $ wasm-pack build --target web --out-name my-halo2-circuit
```

Comment:
we should also let users use `mopro build` 🙏🏻

Reply:
I believe the contents of `rust-setup.md` and `web-wasm-setup.md` are intended for experienced users.
Therefore, I’ve left this section without incorporating the mopro cli. This is a mixed situation, and it might be better to cover how to update a customized circuit using the cli on a separate page.

It could be a good idea to introduce this while working on: https://github.com/zkmopro/mopro/issues/269